tools/cwrap/plugins/__init__.py

```diff
@@ -1,55 +1,418 @@
+        inspect the combined checks and do any additional checking/modify that
+        string as you see fit. In particular, given code is a string like:
+
+        CHECK_TYPE(GET_ARG(0)) && CHECK_TYPE(GET_ARG(1)) && ..
+
+        We can process it as we see fit. For example, we may want to add a
+        check at the beginning that we have the specified number of arguments.
+
+        Args:
+            code: A string representing each argument check separated by an
+            '&&'. code can be None if there are no arguments to be checked.
+            option: dictionary containing the information for this specific
+            option.
+
+        Returns:
+            The modified code string with any additional checks, or just the
+            existing code if no modifications are to be made.
+
+        """
         return code
 
     def process_single_unpack(self, code, arg, arg_accessor):
+        """Used to postprocess a type unpack.
+
+        Same as process_single_check above, but for type unpacking. E.g. an
+        example code could be:
+
+        PyLong_FromLong(PyTuple_GET_ITEM(args, 0))
+
+        And this code could modify that as it sees fit. For example, if the
+        result of accessing the argument is None, we would not want to call the
+        unpacking code.
+
+        Args:
+            code: The string code representing a type unpack for a specific
+            argument being accessed.
+            arg: dictionary containing properties of that specific argument
+            arg_accessor: the arg_accessor string for that specific argument.
+            Note that this is likely also embedded in code, but if you want to
+            be able to access this arg and throw away the other code, you can
+            do so.
+
+        Returns:
+            A string representing the processed unpack/access string for this
+            arg. If the plugin does not know how to modify a specific input, it
+            should return the original code.
+
+        """
         return code
 
     def process_all_call_arg(self, code, option):
+        """Used to modify the arguments to the underlying C function call.
+
+        Code is the string of comma-separated arguments that will be passed to
+        the wrapped C function. You can use this function to modify that string
+        as you see fit. For example, THP prepends the LIBRARY_STATE definition
+        so that the generated code will follow the conventions it uses for
+        writing one function for both TH/THC calls.
+
+        Args:
+            code: A string as described above.
+            option: dictionary containing the information for this specific
+            option.
+
+        Returns:
+            The same code, modified as the plugin sees fit.
+
+        """
         return code
 
     def process_option_code(self, code, option):
+        """Used to modify the entire code body for an option.
```

Comment:
It would be helpful to say that body includes the option checks (the whole `else if (...)`).